### ISSUE TITLE
Optimize For Single Time Usage

### ISSUE THREAD

**Issue by hediet** [NickeManarin/ScreenToGif#977]
First of all: Thanks so much for this nice tool! I use it a lot.

It would be so nice if this tool is more optimized for single-time ad-hoc usage to quickly share a screen recording.

Currently, doing a screenrecording requires many clicks:
1. Start the tool
2. Click on Recorder
3. Click on Record
4. Click on Stop
5. Click on File
6. Click on Save as
7. Click on Save
8. Click on Exit
9. Click on Yes ("Are you sure that you want to exit?")

10. I usually want to post my screen recording in github issues. In this case, I have to open the target file location
11. and drag&drop the gif into the issue editor.

Ideally, my flow would look like this:
1. Start the tool (or press a global hotkey)
2. Click on Record
3. Click on Stop
4. Click on Export
5. Drag & drop an icon in ScreenToGif (that refers to the exported file) into the github issue editor
6. Exit screen to gif without questions

**Comment by pawlos** [comment]
To eliminate step 2 from the original list, you can set-up the application to start in recorder mode (`Application` -> `Start with: Screen recorder`). To eliminate step 10 you can uncheck "Ask me before closing the editor (if there's a project opened)." in `Editor`->`General`. 
For the faster save/export workflow I don't know if there's anything at the moment. 

For future version, I was thinking, maybe it could be done as one of the new automated tasks that run as part of the processing done when the frames are loaded? I think it could fit into the existing framework nicely.  WDYT @NickeManarin? Does it make sense to have Export/Save as one of the automated tasks?

**Comment by NickeManarin** [comment]
The ideal for the near future is to compartimentalize the Editor, separating core features, leaving it possible to bypass it by using a simple workflow:

**Recorder -> Save As**, won't allow any edits at all.
**Recorder -> Simple Editor -> Save As**, will allow playback trimming and perhaps resizing.
**Recorder -> Complete Editor -> Save As**, always set as default.

Automatic tasks should work for all cases, if enabled. 
-------------

Now, what I have in mind to do first is to migrate to .NET 5/6, then start the editor rework/redesign (that is going to take a while), then improve workflow.

The new editor should have support for multiple layers (this will need the creation of a timeline element, I already started), improve rendering (allowing a quality slider like Premier Pro, 100% -> 50% size), separated panels from main view, proper action stack for things like moving layers, altering properties, etc.

The list goes on. I just need time. ;)

**Comment by NickeManarin** [comment]
Btw, the whole project system will work too. The editor will work directly on top of a project file (*.stg), still automatically created (with cache settings still in place).

**Comment by hediet** [comment]
Sounds great!

If you already rework/redesign this tool - what are your thoughts on supporting objects?
Currently, when adding some text, it seems you immediately render the text to all images (which I guess is why it takes so long). Couldn't you just keep track of the text object and render it on demand? That way, I could easily edit and move the text.
Ideally, properties of an object could depend on the time.

**Comment by NickeManarin** [comment]
Yes, that's the main reason for the rework/redesign. To add support for layers and objects, with rendering on remand.
